File: programmers/43165.cpp
#include <string>
#include <vector>
#include <deque>
#include <iostream>

using namespace std;

int result;

void DFS ( deque<int> numbers, int target, int check) {

    if (numbers.empty()) {
        if (check == target) result++;
        return;
    }

    deque<int> numbers2(numbers);

    int temp = check;

    check = check + numbers.front();
    numbers.pop_front();

    DFS(numbers, target, check);

    temp = temp - numbers2.front();
    numbers2.pop_front();

    DFS(numbers2, target, temp);

}

int solution(vector<int> numbers, int target) {
    int answer = 0;
    int check(0);

    deque<int> search;

    for (auto i : numbers) search.push_back(i);

    DFS(search, target, check);

    return result;
}

int main() {
    int check(0), result(0), i(0);
    vector<int> num= {1, 1};
    int target = 0;

    i = solution(num, target);

    cout<<i<<endl;
}

File: programmers/43165.py
def solution(numbers, target):
    return DFS(numbers, target, 0, 0)

def DFS(numbers, target, current, depth):
    if depth == len(numbers):
        if current == target:
            return 1
        else:
            return 0
    else:
        a = DFS(numbers, target, current+numbers[depth], depth+1)
        b = DFS(numbers, target, current-numbers[depth], depth+1)
        return a + b